Automattic/jetpack
# Options: sync mailserver settings

- Sync changes in these options:
  - `mailserver_url`
  - `mailserver_login` — Not syncing contents, only the option name
  - `mailserver_pass`  — Not syncing contents, only the option name
  - `mailserver_port`

Adds new list for contentless options and a new `jetpack_sync_options_contentless` filter for them.

Options listed in the contentless list will be synced but without content. Only the option name is sent.

Instead of:

```
(
    [0] => mailserver_login
    [1] => foo
    [2] => bar
)
```

Syncs this:
```
(
    [0] => mailserver_login
    [1] =>
    [2] =>
)
```

Synced information will be shown in the Activity Log.

`default_email_category` is already synced in `master`.

## Testing

- Modify "Post via email" settings via `/wp-admin/options-writing.php`

    ![image](https://user-images.githubusercontent.com/87168/38865517-ab4a2d42-4246-11e8-844b-79f2cee5a924.png)
- Observe values get synced
- Observe `mailserver_login` and `mailserver_pass` have empty strings as values.
- Tests should pass: `phpunit --filter=WP_Test_Jetpack_Sync_Options`
